setup.py

```diff
@@ -169,34 +169,22 @@ def function_three():
         'License :: OSI Approved :: MIT License',
         'Operating System :: OS Independent',
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12',
         'Topic :: Software Development :: Libraries :: Python Modules',
     ],
     extras_require={
         'dev': [
             'tox',
-            'PyTest             ; python_version >= "3.6"',
-            'PyTest < 5         ; python_version < "3.6"',
-            'PyTest-Cov         ; python_version >= "3.6"',
-            'PyTest-Cov < 2.6   ; python_version < "3.6"',
+            'PyTest',
+            'PyTest-Cov',
             'bump2version < 1',
             'sphinx < 2',
-            # Python 2.7 EOL:
-            'importlib-metadata < 3       ; python_version < "3"',
-            'importlib-resources < 4      ; python_version < "3"',
-            'configparser < 5             ; python_version < "3"',
-            'sphinxcontrib-websupport < 2 ; python_version < "3"',
-            'zipp < 2                     ; python_version < "3"',
         ]
     },
-    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
+    python_requires='>=3.7',
```

Comment:
But surely they could continue using the older releases of this package, if they still rely on python 2.7?

It won't break any user if they are stuck on an older version, because other packages are also not moving and there is no support/security-fixes anyways?

Reply:
This PR does not change the code base, so the code remains compatible with older versions of Python. I don't want to make a major release for this PR: it's only a build fix.

Reply:
If you're OK with allowing installation on an untested Python version, I'll revert the versions drop. I myself would not want to receive reports about any future failure on untested and EOL versions but I recognize the risk of breakage is low given that this library doesn't need to change much.

Reply:
Researching a little deeper, I could retain testing on the older versions by switching the base platform for testing to 

* ubuntu-20.04 for 3.6 & 3.5
* ubuntu-18.04 for 3.4
* simply re-add 2.7 because there's a package for ubuntu-22.04 (which is ubuntu-latest) for 2.7.18

all of these per https://github.com/actions/python-versions/blob/main/versions-manifest.json

Looks like I can also enable `3.12.0-alpha - 3.12.0` through [the semver hyphen ranges](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input).

Reply:
2.7 is being removed in a couple of weeks https://github.com/actions/runner-images/issues/7401

Instead of `3.12.0-alpha - 3.12.0`, we can use the simpler `3.12-dev`. https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input

Reply:
> Researching a little deeper, I could retain testing on the older versions by switching the base platform for testing to 
> 
> 
> 
> * ubuntu-20.04 for 3.6 & 3.5
> 
> * ubuntu-18.04 for 3.4
> 
> * simply re-add 2.7 because there's a package for ubuntu-22.04 (which is ubuntu-latest) for 2.7.18
> 
> 
> 
> all of these per https://github.com/actions/python-versions/blob/main/versions-manifest.json
> 
> 
> 
> Looks like I can also enable `3.12.0-alpha - 3.12.0` through [the semver hyphen ranges](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input).

I am OK with that.